main.tf

```
@@ -0,0 +1,192 @@
+data "aws_partition" "current" {}
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "lambda" {
+  statement {
+    sid = "AllowS3Object"
+    actions = [
+      "s3:PutObject",
+      "s3:PutObjectTagging",
+      "s3:PutObjectVersionTagging",
+    ]
+    resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.s3_bucket}/*"]
+  }
+
+  statement {
+    actions = [
+      "ses:SendEmail",
+      "ses:SendTemplatedEmail",
+      "ses:TestRenderTemplate"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+
+  statement {
+    sid = "AllowAssumeRole"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:iam::*:role/${var.assume_role_name}"
+    ]
+  }
+}
+
+module "lambda" {
+  source = "git::https://github.com/terraform-aws-modules/terraform-aws-lambda.git?ref=v7.2.1"
+
+  description   = "Lambda function for Key Enforcement"
+  function_name = var.project_name
+  handler       = "iam_key_enforcer.lambda_handler"
+  tags          = var.tags
+
+  attach_policy_json = true
+  policy_json        = data.aws_iam_policy_document.lambda.json
+
+  artifacts_dir            = var.lambda.artifacts_dir
+  build_in_docker          = var.lambda.build_in_docker
+  create_package           = var.lambda.create_package
+  ignore_source_code_hash  = var.lambda.ignore_source_code_hash
+  local_existing_package   = var.lambda.local_existing_package
+  recreate_missing_package = var.lambda.recreate_missing_package
+  ephemeral_storage_size   = var.lambda.ephemeral_storage_size
+  runtime                  = var.lambda.runtime
+  s3_bucket                = var.lambda.s3_bucket
+  s3_existing_package      = var.lambda.s3_existing_package
+  s3_prefix                = var.lambda.s3_prefix
+  store_on_s3              = var.lambda.store_on_s3
+  timeout                  = var.lambda.timeout
+
+  environment_variables = {
+    LOG_LEVEL                  = var.log_level
+    EMAIL_ADMIN_REPORT_ENABLED = var.email_admin_report_enabled
+    EMAIL_ADMIN_REPORT_SUBJECT = var.email_admin_report_subject
+    EMAIL_SOURCE               = var.email_source
+    ADMIN_EMAIL                = var.admin_email
+    KEY_AGE_WARNING            = var.key_age_warning
+    KEY_AGE_INACTIVE           = var.key_age_inactive
+    KEY_AGE_DELETE             = var.key_age_delete
+    KEY_USE_THRESHOLD          = var.key_use_threshold
+    S3_ENABLED                 = var.s3_enabled
+    S3_BUCKET                  = var.s3_bucket
+    EMAIL_TAG                  = var.email_tag
+    EMAIL_BANNER_MSG           = var.email_banner_message
+    EMAIL_BANNER_MSG_COLOR     = var.email_banner_message_color
+    EMAIL_USER_TEMPLATE        = aws_ses_template.user_template.id
+    EMAIL_ADMIN_TEMPLATE       = aws_ses_template.admin_template.id
+  }
+
+  source_path = [
+    {
+      path          = "${path.module}/src/python",
+      prefix_in_zip = ""
+      patterns      = var.lambda.source_path.patterns
+    },
```

Add pip_requirements to this block
```suggestion
    {
      path          = "${path.module}/src/python",
      pip_requirements = true
      prefix_in_zip = ""
      patterns      = var.lambda.source_path.patterns
    },
```